posixlib/src/main/resources/scala-native/time.c

```diff
@@ -31,6 +34,18 @@ static void scalanative_tm_init(struct scalanative_tm *scala_tm,
 }
 
 static void tm_init(struct tm *tm, struct scalanative_tm *scala_tm) {
+  if (sizeof(struct tm) > sizeof(struct scalanative_tm)) {
+        // The size of the operating system struct tm can be larger than
+        // the scalanative tm.  On Linux this is true.
+        // Clearing the entire tm and then setting known fields ensures
+        // that any fields not known to scalanative, such as tm_zone,
+	// are zero/NULL, not J-Random garbage.
+	// strftime() in Scala Native release mode is particularly sensitive to
+	// garbage beyond the end of the scalanative tm.
+
+        memset(tm, 0, sizeof(*tm));
```

Comment:
as far as I can see `tm` is always allocated on the stack. Maybe we can use `{0}` to ask compiler to allocate in with zeros and remove `if` and `memset` from runtime?

Reply:
I took one run of consideration for this topic in the current evolution.  Let me give it some
thought overnight.  I do not like the mumble_init functions, but it puts the initialization
in one place, and does not rely upon devos adding `{0}` to each new function: an error prone process.

Perhaps I will be more clever by the light of day, or after the Solstice.

Thank you for collegialy pushing for clarity.